src/config.rs

```diff
@@ -199,7 +198,14 @@ fn parse_line(dev: &mut Device, key: &str, value: &str) -> Result<()> {
         "zram-fraction" => {
```

Comment:
@keszybz: you implemented the thing above:
```rust
    for dir in base_dirs.iter().rev() {
        let path = PathBuf::from(dir).join("systemd/zram-generator.conf");
        if path.exists() {
            fragments.insert("".to_string(), path); // The empty string shall sort earliest
            break;
        }
    }
```

This feels broken now that I look at it: `fragments.insert()` will replace all config files with the latest one it finds (in `/usr/lib`). Is this what you intended?

Reply:
IIUC yes. it has rev() so it should go from /usr down to /run so the first one wins. aka /etc will override /usr/lib completely.

Reply:
Given
```rust
    let base_dirs = vec![
        String::from(root.join("usr/lib").to_str().unwrap()),
        String::from(root.join("usr/local/lib").to_str().unwrap()),
        String::from(root.join("etc").to_str().unwrap()),
        String::from(root.join("run").to_str().unwrap()), // We look at /run to allow temporary overriding
                                                          // of configuration. There is no expectation of
                                                          // programatic creation of config there.
    ];
```

`base_dirs.rev()` will yield `["/usr/lib", "/usr/local/lib", "/etc", "/run"]`

Reply:
Given
```rust
    let base_dirs = vec![
        String::from(root.join("usr/lib").to_str().unwrap()),
        String::from(root.join("usr/local/lib").to_str().unwrap()),
        String::from(root.join("etc").to_str().unwrap()),
        String::from(root.join("run").to_str().unwrap()), // We look at /run to allow temporary overriding
                                                          // of configuration. There is no expectation of
                                                          // programatic creation of config there.
    ];
```

`base_dirs.rev()` will yield `["/run", "/etc", "/usr/local/lib", "/usr/lib"]` => `/usr/lib/...`, if found, will override `/run/...`, if found

Reply:
Ugh sorry disregard this, I missed the `break;`.